extensions/wikidata/src/org/openrefine/wikidata/qa/scrutinizers/ItemRequiresScrutinizer.java

```diff
@@ -62,27 +63,26 @@ public void scrutinize(ItemUpdate update) {
             }
         }
 
-        for(PropertyIdValue propertyId : propertyIdValueValueMap.keySet()){
+        for (PropertyIdValue propertyId : propertyIdValueValueMap.keySet()) {
             List<Statement> constraintDefinitions = _fetcher.getConstraintsByType(propertyId, ITEM_REQUIRES_CONSTRAINT_QID);
             for (Statement statement : constraintDefinitions) {
                 ItemRequiresConstraint constraint = new ItemRequiresConstraint(statement);
                 PropertyIdValue itemRequiresPid = constraint.itemRequiresPid;
                 List<Value> itemList = constraint.itemList;
                 if (!propertyIdValueValueMap.containsKey(itemRequiresPid)) {
```

Comment:
If I am not wrong, the two branches of this `if { } else if {}` are identical, so you could deduplicate the code with an OR (`||`) in your first `if`. But because of that, I am wondering if the `else if` branch should not be different: we probably want to display a different message for such a violation. If property A requires property B with a list of allowed values, and we have a violation in the `else if` branch here, we should say that although there is indeed a statement with property B, it does not have one of the allowed values, hence the violation. That probably means having 4 different violation messages for this constraint in total!

Reply:
I'm not having a clear idea about this, how can we have 4 different violation messages?
Property A requires property B with a set of allowed values, then the violations will be: 
1. It doesn't have property B, then we don't have to check about the values
2. It does have property B but it doesn't have the required value.

Can you specific the other two?

Reply:
Sure:
1. It doesn't have property B, and we don't have to check about the values. And the item where the constraint is found already exists.
2. It does have property B but it doesn't have the required value. And the item where the constraint is found already exists.
3. It doesn't have property B, and we don't have to check about the values. And the item where the constraint is found is new.
4. It does have property B but it doesn't have the required value. And the item where the constraint is found is new.